### PR TITLE
Optional github actions setup

### PR DESCRIPTION
**Pull request by adelrodriguez**
Add an optional GitHub Actions workflow setup during `init` to automate running enabled checks and replace the `ci` command.

---
<a href="https://cursor.com/background-agent?bcId=bc-2b3c3e3f-e49a-4ea9-ab5b-6ab99922dfb2"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-cursor-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-cursor-light.svg">Open in Cursor</picture></a>&nbsp;<a href="https://cursor.com/agents?id=bc-2b3c3e3f-e49a-4ea9-ab5b-6ab99922dfb2"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-web-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-web-light.svg">Open in Web</picture></a>

## Summary by CodeRabbit

* **Breaking Changes**
  * Removed the built-in CI command from the CLI.

* **New Features**
  * Optional GitHub Actions workflow setup during project initialization.
  * Automatic package-manager detection and tailored workflow steps.
  * Generated workflow runs enabled checks with caching and concurrency; replaces separate CI invocation.

* **Tests**
  * Added integration tests for the workflow helper and CI-script detection.
  * Removed CI-specific unit tests tied to the removed CI command.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by adelrodriguez — COMMENTED** [review]
@cursor also add tests for this PR

File: src/commands/init.ts [hunk 1]
```diff
@@ -185,6 +186,27 @@ const setupEditors = (editors: string[]) =>
     return ok()
   })
 
+const setupGitHubActions = (
+  packageManager: "npm" | "yarn" | "pnpm" | "bun",
```
[the remark this answers is not in the document]
Comment (adelrodriguez) [reply src/commands/init.ts]: @cursor do not exclude deno

File: src/commands/init.ts [hunk 2]
```diff
@@ -305,6 +338,10 @@ export default defineCommand({
 
       yield* setupEditors(editors)
 
+      if (enableGitHubActions) {
+        yield* setupGitHubActions(packageManager as "npm" | "yarn" | "pnpm" | "bun", scripts)
+      }
```
[the remark this answers is not in the document]
Comment (adelrodriguez) [reply src/commands/init.ts]: @cursor should we address this?

File: src/helpers/ci/github.ts [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+const generateWorkflow = ({ packageManager, scripts }: WorkflowOptions): string | null => {
+  const jobs: string[] = []
+
+  // Map scripts to jobs
+  if (scripts.includes("check")) {
+    jobs.push(generateJob("lint", "Run linter", "check", packageManager))
+  }
+
+  if (scripts.includes("format")) {
+    jobs.push(generateJob("format", "Check formatting", "format", packageManager))
```
Comment (adelrodriguez) [inline src/helpers/ci/github.ts]: @cursor this should be `format --check`

**Comment by adelrodriguez** [comment]
> **Actionable comments posted: 0**
> 
> <details>
> <summary>🧹 Nitpick comments (2)</summary><blockquote>
> 
> <details>
> <summary>src/helpers/ci/github.ts (1)</summary><blockquote>
> 
> `120-127`: **Consider extracting CI scripts as a shared constant.**
> 
> The CI-compatible scripts list (`["check", "format", "typecheck", "check:monorepo"]`) is implicitly duplicated between `hasCICompatibleScripts` and the `generateWorkflow` function's conditional checks. Extracting this as a module-level constant would ensure they stay in sync.
> 
> 
> <details>
> <summary>🔎 Suggested approach</summary>
> 
> ```diff
> +const CI_COMPATIBLE_SCRIPTS = ["check", "format", "typecheck", "check:monorepo"] as const
> +
>  export const hasCICompatibleScripts = (scripts: string[]): boolean => {
> -  const ciScripts = ["check", "format", "typecheck", "check:monorepo"]
> -  return scripts.some((script) => ciScripts.includes(script))
> +  return scripts.some((script) => CI_COMPATIBLE_SCRIPTS.includes(script as typeof CI_COMPATIBLE_SCRIPTS[number]))
>  }
> ```
> </details>
> 
> </blockquote></details>
> <details>
> <summary>__tests__/helpers.integration.test.ts (1)</summary><blockquote>
> 
> `536-690`: **Good test coverage for the GitHub helper.**
> 
> The tests thoroughly cover workflow generation for major package managers (bun, npm, pnpm, yarn), error handling scenarios, and edge cases like empty scripts and existing workflows.
> 
> 
> 
> **Optional:** Consider adding a test case for `deno` package manager since it's included in `setupSteps` but not exercised by any test.
> 
> 
> <details>
> <summary>🔎 Example test for Deno</summary>
> 
> ```typescript
> test("should generate correct workflow for deno", async () => {
>   const createResult = await github.create({
>     packageManager: "deno",
>     scripts: ["check"],
>   })
>   createResult._unsafeUnwrap()
> 
>   const content = await Bun.file(".github/workflows/adamantite.yml").text()
>   expect(content).toContain("Setup Deno")
>   expect(content).toContain("denoland/setup-deno@v2")
>   expect(content).toContain("deno install --frozen")
> })
> ```
> </details>
> 
> </blockquote></details>
> 
> </blockquote></details>
> 
> <details>
> <summary>📜 Review details</summary>
> 
> **Configuration used**: defaults
> 
> **Review profile**: CHILL
> 
> **Plan**: Pro
> 
> <details>
> <summary>📥 Commits</summary>
> 
> Reviewing files that changed from the base of the PR and between 11a83a45ad87dc3443975fb665f75f2621bf31ae and f21c7d3402d7562f9fca472bb138b90922401a87.
> 
> </details>
> 
> <details>
> <summary>📒 Files selected for processing (3)</summary>
> 
> * `__tests__/helpers.integration.test.ts`
> * `src/commands/init.ts`
> * `src/helpers/ci/github.ts`
> 
> </details>
> 
> <details>
> <summary>🚧 Files skipped from review as they are similar to previous changes (1)</summary>
> 
> * src/commands/init.ts
> 
> </details>
> 
> <details>
> <summary>🧰 Additional context used</summary>
> 
> <details>
> <summary>📓 Path-based instructions (1)</summary>
> 
> <details>
> <summary>**/*.{ts,tsx}</summary>
> 
> 
> **📄 CodeRabbit inference engine (AGENTS.md)**
> 
> > Perform type checking using `bun run typecheck`
> 
> Files:
> - `src/helpers/ci/github.ts`
> - `__tests__/helpers.integration.test.ts`
> 
> </details>
> 
> </details><details>
> <summary>🧠 Learnings (12)</summary>
> 
> <details>
> <summary>📓 Common learnings</summary>
> 
> ```
> Learnt from: CR
> Repo: adelrodriguez/adamantite PR: 0
> File: AGENTS.md:0-0
> Timestamp: 2025-12-25T04:04:41.286Z
> Learning: Applies to **/src/commands/ci.ts : Implement ci command in `src/commands/ci.ts` for running Adamantite in CI environments
> ```
> 
> ```
> Learnt from: CR
> Repo: adelrodriguez/adamantite PR: 0
> File: AGENTS.md:0-0
> Timestamp: 2025-12-25T04:04:41.286Z
> Learning: Applies to **/src/commands/init.ts : Implement init command in `src/commands/init.ts` to initialize Adamantite configuration
> ```
> 
> ```
> Learnt from: CR
> Repo: adelrodriguez/adamantite PR: 0
> File: AGENTS.md:0-0
> Timestamp: 2025-12-25T04:04:41.286Z
> Learning: Applies to **/src/commands/update.ts : Implement update command in `src/commands/update.ts` to update Adamantite configuration
> ```
> 
> </details>
> <details>
> <summary>📚 Learning: 2025-12-25T04:04:41.286Z</summary>
> 
> ```
> Learnt from: CR
> Repo: adelrodriguez/adamantite PR: 0
> File: AGENTS.md:0-0
> Timestamp: 2025-12-25T04:04:41.286Z
> Learning: Applies to **/src/commands/ci.ts : Implement ci command in `src/commands/ci.ts` for running Adamantite in CI environments
> ```
> 
> **Applied to files:**
> - `src/helpers/ci/github.ts`
> - `__tests__/helpers.integration.test.ts`
> 
> </details>
> <details>
> <summary>📚 Learning: 2025-12-25T04:04:41.286Z</summary>
> 
> ```
> Learnt from: CR
> Repo: adelrodriguez/adamantite PR: 0
> File: AGENTS.md:0-0
> Timestamp: 2025-12-25T04:04:41.286Z
> Learning: Applies to **/src/utils.ts : Place shared utilities in `src/utils.ts` for package manager detection and error handling
> ```
> 
> **Applied to files:**
> - `src/helpers/ci/github.ts`
> - `__tests__/helpers.integration.test.ts`
> 
> </details>
> <details>
> <summary>📚 Learning: 2025-12-25T04:04:41.286Z</summary>
> 
> ```
> Learnt from: CR
> Repo: adelrodriguez/adamantite PR: 0
> File: AGENTS.md:0-0
> Timestamp: 2025-12-25T04:04:41.286Z
> Learning: Applies to **/src/commands/monorepo.ts : Implement monorepo command in `src/commands/monorepo.ts` with `--fix` flag to auto-fix issues in monorepo environments
> ```
> 
> **Applied to files:**
> - `src/helpers/ci/github.ts`
> 
> </details>
> <details>
> <summary>📚 Learning: 2025-12-25T04:04:41.286Z</summary>
> 
> ```
> Learnt from: CR
> Repo: adelrodriguez/adamantite PR: 0
> File: AGENTS.md:0-0
> Timestamp: 2025-12-25T04:04:41.286Z
> Learning: Applies to **/*.{ts,tsx,html,css,js,jsx},**/package.json : Use Bun instead of Node.js, npm, pnpm, or vite for package management and script execution
> ```
> 
> **Applied to files:**
> - `__tests__/helpers.integration.test.ts`
> 
> </details>
> <details>
> <summary>📚 Learning: 2025-12-25T04:04:41.286Z</summary>
> 
> ```
> Learnt from: CR
> Repo: adelrodriguez/adamantite PR: 0
> File: AGENTS.md:0-0
> Timestamp: 2025-12-25T04:04:41.286Z
> Learning: Applies to **/src/index.ts,**/bunup.config.ts : Build CLI using `bun run build` which bundles `src/index.ts` to `dist/` using bunup
> ```
> 
> **Applied to files:**
> - `__tests__/helpers.integration.test.ts`
> 
> </details>
> <details>
> <summary>📚 Learning: 2025-12-25T04:04:41.286Z</summary>
> 
> ```
> Learnt from: CR
> Repo: adelrodriguez/adamantite PR: 0
> File: AGENTS.md:0-0
> Timestamp: 2025-12-25T04:04:41.286Z
> Learning: Applies to **/bunup.config.ts : Use bunup for bundling CLI with minification, configured in `bunup.config.ts`
> ```
> 
> **Applied to files:**
> - `__tests__/helpers.integration.test.ts`
> 
> </details>
> <details>
> <summary>📚 Learning: 2025-12-25T04:04:41.286Z</summary>
> 
> ```
> Learnt from: CR
> Repo: adelrodriguez/adamantite PR: 0
> File: AGENTS.md:0-0
> Timestamp: 2025-12-25T04:04:41.286Z
> Learning: Applies to **/package.json : Install dependencies using `bun install`
> ```
> 
> **Applied to files:**
> - `__tests__/helpers.integration.test.ts`
> 
> </details>
> <details>
> <summary>📚 Learning: 2025-12-25T04:04:41.286Z</summary>
> 
> ```
> Learnt from: CR
> Repo: adelrodriguez/adamantite PR: 0
> File: AGENTS.md:0-0
> Timestamp: 2025-12-25T04:04:41.286Z
> Learning: Applies to **/*.{ts,tsx} : Perform type checking using `bun run typecheck`
> ```
> 
> **Applied to files:**
> - `__tests__/helpers.integration.test.ts`
> 
> </details>
> <details>
> <summary>📚 Learning: 2025-12-25T04:04:41.286Z</summary>
> 
> ```
> Learnt from: CR
> Repo: adelrodriguez/adamantite PR: 0
> File: AGENTS.md:0-0
> Timestamp: 2025-12-25T04:04:41.286Z
> Learning: Applies to **/*.test.{ts,tsx},**/*.spec.{ts,tsx} : Run tests using `bun test` or `bun run test:watch` for watch mode
> ```
> 
> **Applied to files:**
> - `__tests__/helpers.integration.test.ts`
> 
> </details>
> <details>
> <summary>📚 Learning: 2025-12-25T04:04:41.286Z</summary>
> 
> ```
> Learnt from: CR
> Repo: adelrodriguez/adamantite PR: 0
> File: AGENTS.md:0-0
> Timestamp: 2025-12-25T04:04:41.286Z
> Learning: Applies to **/src/commands/check.ts : Implement check command in `src/commands/check.ts` to run Biome linter
> ```
> 
> **Applied to files:**
> - `__tests__/helpers.integration.test.ts`
> 
> </details>
> <details>
> <summary>📚 Learning: 2025-12-25T04:04:41.286Z</summary>
> 
> ```
> Learnt from: CR
> Repo: adelrodriguez/adamantite PR: 0
> File: AGENTS.md:0-0
> Timestamp: 2025-12-25T04:04:41.286Z
> Learning: Version locally using `bun run version` to bump package.json and update CHANGELOG before publishing
> ```
> 
> **Applied to files:**
> - `__tests__/helpers.integration.test.ts`
> 
> </details>
> 
> </details><details>
> <summary>🧬 Code graph analysis (2)</summary>
> 
> <details>
> <summary>src/helpers/ci/github.ts (1)</summary><blockquote>
> 
> <details>
> <summary>src/utils.ts (1)</summary>
> 
> * `checkIfExists` (53-57)
> 
> </details>
> 
> </blockquote></details>
> <details>
> <summary>__tests__/helpers.integration.test.ts (1)</summary><blockquote>
> 
> <details>
> <summary>src/helpers/ci/github.ts (2)</summary>
> 
> * `github` (129-191)
> * `hasCICompatibleScripts` (124-127)
> 
> </details>
> 
> </blockquote></details>
> 
> </details>
> 
> </details>
> 
> <details>
> <summary>🔇 Additional comments (7)</summary><blockquote>
> 
> <details>
> <summary>src/helpers/ci/github.ts (5)</summary><blockquote>
> 
> `1-13`: **LGTM!**
> 
> Clean imports and type usage. Good adoption of `PackageManagerName` from `nypm` as per prior feedback, ensuring type alignment with the package manager detection library.
> 
> ---
> 
> `59-75`: **LGTM!**
> 
> Clean job generation with proper YAML indentation. Good use of `runScriptCommand` from `nypm` to correctly format script execution per package manager, and sensible defaults for `ubuntu-latest` and `timeout-minutes: 10`.
> 
> ---
> 
> `77-118`: **LGTM!**
> 
> The workflow generation correctly handles empty jobs by returning `null`, and properly uses `format --check` for CI formatting verification. The concurrency settings with `cancel-in-progress: true` are a good practice to avoid redundant workflow runs.
> 
> ---
> 
> `170-190`: No action needed. The `update()` method is only called after `github.exists()` returns true (in init.ts line 195-197), which checks if the workflow file exists at `.github/workflows/adamantite.yml`. If the file exists, the directory logically must exist, so the defensive directory check is unnecessary.
> 
> 
> > Likely an incorrect or invalid review comment.
> 
> ---
> 
> `15-57`: No issue. The `setupSteps` Record covers all possible `PackageManagerName` values from nypm (npm, yarn, pnpm, bun, deno). TypeScript's `Record<PackageManagerName, string>` type enforces exhaustiveness at compile time, so missing keys would result in a compile error.
> 
> 
> > Likely an incorrect or invalid review comment.
> 
> </blockquote></details>
> <details>
> <summary>__tests__/helpers.integration.test.ts (2)</summary><blockquote>
> 
> `6-6`: **LGTM!**
> 
> Clean import addition for the new GitHub helper utilities.
> 
> ---
> 
> `692-720`: **LGTM!**
> 
> Comprehensive test coverage for `hasCICompatibleScripts` including all individual CI scripts, negative cases (fix-only scripts, empty array), and mixed scenarios.
> 
> </blockquote></details>
> 
> </blockquote></details>
> 
> </details>
> 
> 

@cursor address the nitpick comments
